notes_app/client/notes.js

```diff
@@ -82,18 +82,9 @@
 
   angular.module('notes').config(config);
 
-  angular.module('notes').run(function($rootScope, $state, userService, $q) {
+  angular.module('notes').run(function($rootScope, $state, userService) {
     $rootScope.$on('$stateChangeStart',
-      function(event, toState, toParams, fromState, fromParams, options) {
-        userService.checkSession().then(function(session){
-          if (toState.data && toState.data['loginRequired']) {
-            if (session.data.id == undefined) {
-              event.preventDefault();
-              $state.go('main.welcome');
-            }
-          }
-        })
-      });
+      function(event, toState, toParams, fromState, fromParams, options) {});
```

Comment:
Is that an empty callback? It probably could be removed along with the whole run block.